_source/_posts/2019-10-21-illustrated-guide-to-oauth-and-oidc.md

```diff
@@ -0,0 +1,158 @@
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/client-id.jpg alt:"Client ID" width:"200" %}</td>
+    <td markdown="span">**Client ID**: This ID is used to identify the **Client** with the **Authorization Server**.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/client-secret.jpg alt:"Client Secret" width:"200" %}</td>
+    <td markdown="span">**Client Secret**: This is a secret password that only the **Client** and **Authorization Server** know. This allows them to securely share information privately behind the scenes.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/authorization-code.jpg alt:"Authorization Code" width:"200" %}</td>
+    <td markdown="span">**Authorization Code**: A short-lived temporary code the **Client** gives the **Authorization Server** in exchange for an **Access Token**.</td>
+</tr>
+<tr>
+    <td>{% img blog/illustrated-guide-to-oauth-and-oidc/access-token.jpg alt:"Access Token" width:"200" %}</td>
+    <td markdown="span">**Access Token**: The key the client will use to communicate with the **Resource Server**. This is like a badge or key card that gives the **Client** permission to request data or perform actions with the **Resource Server** on your behalf.</td>
+</tr>
+</table>
+
+> Note: Sometimes the "Authorization Server" and the "Resource Server" are the same server. However, there are cases where they will _not_ be the same server or even part of the same organization. For example, the "Authorization Server" might be a third-party service the "Resource Server" trusts.
+
+Now that we have some of the OAuth 2.0 vocabulary handy, let's revisit the example with a closer look at what's going on throughout the OAuth flow.
+
+{% img blog/illustrated-guide-to-oauth-and-oidc/tpotd-authorization-code-flow.jpg alt:"Terrible Pun of the Day Authorization Code Flow" width:"800" %}
+
+1. You, the **Resource Owner**, want to allow "Terrible Pun of the Day," the **Client**, to access your contacts so they can send invitations to all your friends.
+1. The **Client** redirects your browser to the **Authorization Server** and includes with the request the **Client ID**, **Redirect URI**, **Response Type**, and one or more **Scopes** it needs.
+1. The **Authorization Server** verifies who you are, and if necessary prompts for a login.
+1. The **Authorization Server** presents you with a **Consent** form based on the **Scopes** requested by the **Client**. You grant (or deny) permission.
+1. The **Authorization Server** redirects back to **Client** using the **Redirect URI** along with an **Authorization Code**.
+1. The **Client** contacts the **Authorization Server** directly (does not use the **Resource Owner**'s browser) and securely sends its **Client ID**, **Client Secret**, and the **Authorization Code**.
```

Comment:
`grant_type=authorization_code` is required at this step too. Maybe too much detail? Not a show stopper...